File: WildWatchAPI/Models/Species.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Driver;
using System.Text.Json.Serialization;

namespace WildWatchAPI.Models
{
    public class SpeciesSummary
    {
        public string speciesId { get; set; } = null!;
        [JsonConverter(typeof(JsonStringEnumConverter))]
        [BsonRepresentation(BsonType.String)]
        public Klasa Class { get; set; }
        public string CommonName { get; set; } = null!;
        public string ScientificName { get; set; } = null!;
        public string ImageUrl { get; set; } = null!;
        public string Description { get; set; } = null!;
        public string ConservationStatus { get; set; } = null!;
    }
    public class Species
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }
        [JsonConverter(typeof(JsonStringEnumConverter))]
        [BsonRepresentation(BsonType.String)]
        public Klasa Class { get; set; }
        public string CommonName { get; set; } = null!;
        public string ScientificName { get; set; } = null!;
        public string ImageUrl { get; set; } = null!;
        public string Description { get; set; }= null!;
        public string ConservationStatus { get; set; }= null!;

        //[JsonIgnore]
        public List<SightingSummarySpecies> Sightings { get; set; } = new();

        //[JsonIgnore]
        public List<HabitatSummary> Habitats { get; set; } = new();


    }
    public enum Klasa
    {
        Sponge,  //sundjeri
        Cnidaria, // dupljari
        Platyhelminthes,//pljosnati crvi
        Annelida, //clankoviti crvi
        Mollusca, //mekusci
        Cephalopoda,//glavonosci
        Arthropoda, //zglavkari
        Echinodermata, //bodljokosci
        Tunicata, //plastasi
        Cephalochordata, //kopljasi
        Fish, //ribe
        Amphibians, //vodozemci
        Reptilia, //gmizavci
        Birds, //ptice
        Mammals// sisari
    }
}

File: Frontend/wild-watch-frontend/src/models/Species.ts
import { HabitatSummary } from "./Habitat"
import { SightingSummarySpecies } from "./Sighting"

export class SpeciesSummary {
    speciesId: string
    speciesClass: Klasa
    commonName: string
    scientificName: string
    imageUrl: string
    description: string
    conservationStatus: string

    constructor(
        id: string,
        classAnimal: Klasa,
        commonName: string,
        scientificName: string,
        imageUrl: string,
        description: string,
        conservationStatus: string
    ) {
        this.speciesId = id;
        this.speciesClass = classAnimal;
        this.commonName = commonName;
        this.scientificName = scientificName;
        this.imageUrl = imageUrl;
        this.description = description;
        this.conservationStatus = conservationStatus;
    }

}


export class SpeciesDto {
    id: string
    speciesClass: Klasa
    commonName: string
    scientificName: string
    imageUrl: string
    description: string
    conservationStatus: string

    constructor(
        id: string,
        speciesClass: Klasa,
        commonName: string,
        scientificName: string,
        imageUrl: string,
        description: string,
        conservationStatus: string
    ) {
        this.id = id;
        this.speciesClass = speciesClass;
        this.commonName = commonName;
        this.scientificName = scientificName;
        this.imageUrl = imageUrl;
        this.description = description;
        this.conservationStatus = conservationStatus;
    }

}


export class Species {
    id: string
    speciesClass: Klasa
    commonName: string
    scientificName: string
    imageUrl: string
    description: string
    conservationStatus: string
    sightings: SightingSummarySpecies[]
    habitats: HabitatSummary[]

    constructor(
        id: string,
        speciesClass: Klasa,
        commonName: string,
        scientificName: string,
        imageUrl: string,
        description: string,
        conservationStatus: string,
        sightings: SightingSummarySpecies[],
        habitats: HabitatSummary[]
    ) {
        this.id = id;
        this.speciesClass = speciesClass;
        this.commonName = commonName;
        this.scientificName = scientificName;
        this.imageUrl = imageUrl;
        this.description = description;
        this.conservationStatus = conservationStatus;
        this.sightings = sightings;
        this.habitats = habitats;
    }
}

export class FavouriteSpecies{
    public id: string;
    public commonName: string ;

    constructor(id:string, commonName: string){
        this.id = id
        this.commonName = commonName
    }
}

export enum Klasa {
    Sponge,  //sundjeri
    Cnidaria, // dupljari
    Platyhelminthes,//pljosnati crvi
    Annelida, //clankoviti crvi
    Mollusca, //mekusci
    Cephalopoda,//glavonosci
    Arthropoda, //zglavkari
    Echinodermata, //bodljokosci
    Tunicata, //plastasi
    Cephalochordata, //kopljasi
    Fish, //ribe
    Amphibians, //vodozemci
    Reptilia, //gmizavci
    Birds, //ptice
    Mammals// sisari
}

export const KlasaInvert: { [key: string]: Klasa } = {
    "Sponge": Klasa.Sponge,
    "Cnidaria": Klasa.Cnidaria,
    "Platyhelminthes": Klasa.Platyhelminthes,
    "Annelida": Klasa.Annelida,
    "Mollusca": Klasa.Mollusca,
    "Cephalopoda": Klasa.Cephalopoda,
    "Arthropoda": Klasa.Arthropoda,
    "Echinodermata": Klasa.Echinodermata,
    "Tunicata": Klasa.Tunicata,
    "Cephalochordata": Klasa.Cephalochordata,
    "Fish": Klasa.Fish,
    "Amphibians": Klasa.Amphibians,
    "Reptilia": Klasa.Reptilia,
    "Birds": Klasa.Birds,
    "Mammals": Klasa.Mammals
};